Source/Core/Core/State.cpp

```diff
@@ -64,7 +65,7 @@ static AfterLoadCallbackFunc s_on_after_load_callback;
 // Temporary undo state buffer
 static std::vector<u8> g_undo_load_buffer;
 static std::vector<u8> g_current_buffer;
-static bool s_load_or_save_in_progress;
+static Common::Flag s_load_or_save_in_progress;
```

Comment:
What do you think about making this a mutex that we try_lock? That way it's harder to forget to unlock it. (The current solution is fairly good too, so if you think it makes more sense the way it is right now, I'm fine with that.)

Reply:
That's probably not a bad idea. Let's see...

Reply:
Like this?